pippo-core/src/main/java/ro/pippo/core/Pippo.java

```diff
@@ -245,7 +245,10 @@ public void onEvent(ReloadWatcher.Event event, Path dir, Path path) {
     }
 
     private void addShutdownHook() {
-        Runtime.getRuntime().addShutdownHook(new Thread(Pippo.this::stop));
```

Comment:
I think that we should preserve the original line (more compact) because the check of `running` flag is done in `stop()`.